blog/blog.md

```diff
@@ -0,0 +1,265 @@
+
+## Atomic Transaction Composer
+
+Using the [Atomic Transaction Composer](https://developer.algorand.org/docs/get-details/atc/) and the the ABI spec for your contract, you can easily compose atomic group transactions and have the arguments encoded and return values decoded for you!
+
+## Pyteal ABI
+
+PyTeal now provides the necessary components to handle encoding/decoding of types in a contract.  The PyTeal `Router` class even provides a way to route method handling logic and passing decoded types directly to a method as well as providing the ABI contract spec.
+
+For example, if you want to create and handle a method that adds 1 to a uint8 you can do so thusly: 
+
+```py
+@router.method
+def increment_my_number(input: abi.Uint8, *, output: abi.Uint8):
+    return output.set(input.get() + Int(1))
+```
+
+<img src="face_touch.png" alt="facetouch" style="width:100px">
+
+So. Much. Nicer.
+
+For more background see the blog post [here](https://medium.com/algorand/pyteal-introduces-abi-support-for-smart-contracts-605153e91c5e)
+
+For detailed docs on PyTeal ABI see docs [here](https://pyteal.readthedocs.io/en/stable/abi.html)
+
+## Source Maps
+
+Getting a `pc` returned from an algod error message meant you had to assemble, then disassemble your teal contract to find the TEAL line it might be associated to. Even then the names and formatting is mangled in the process so it was hard to track exactly where you were in your program.
+
+You can now compile TEAL with the `sourcemap` flag [enabled](https://developer.algorand.org/docs/rest-apis/algod/v2/#post-v2tealcompile).  The resulting map comes back according to [this spec](https://sourcemaps.info/spec.html) and can be decoded with any of the SDKs using the new `SourceMap` object.
+
+This means you can associate a `pc` directly to the source TEAL line with all the familiar names and formatting you're used to looking at.
+
+
+# Hello Beaker
+
+Today we are sharing [Beaker](https://github.com/algorand-devrel/beaker), a Smart Contract development framework meant to improve the development experience. 
+
+Beaker takes advantage of the above improvements, allowing us to provide much more structure to applications. 
+
+Heads up though, it is still experimental.
+
+<img src="beaker_fire.jpg" alt="beaker fire" style="width:300px">
+
+
+Full Docs are [here](https://algorand-devrel.github.io/beaker/html/index.html).
+
+Taking the above issues, lets see how Beaker helps.
+
+## Code Organization
+
+Beaker provides standard way to organize code using a class to encapsulate functionality.
+
+```py
+from beaker import Application, external
+
+class MyApp(Application):
+    @external
+    def add(self, a: abi.Uint64, b: abi.Uint64, *, output: abi.Uint64):
+        return output.set(a.get() + b.get())
+```
+
+This is a full application! It's got an `approval program`, `clear program`, an implicitly empty `state` and provides the ABI `contract` to export for other clients.
```

Comment:
what do you mean by "and provides the ABI `contract` to export for other clients"?